### PR TITLE
Feat seqtk update

### LINKED ISSUE #502: Expand seqtk module
## Is your feature request related to a problem? Please describe

1) The current seqtk module does not have an implementation of the `subseq` utility from seqtk.

2) Also, the `sample` process does not check if the seed is fixed when sampling paired-end reads (this is necessary to keep reads paired, see [seqtk readme](https://github.com/lh3/seqtk)).

## Describe the solution you'd like

I am working on this functionality right now and going to submit a pull request.

### PR DESCRIPTION
**Pull request by sidorov-si**
## PR checklist

Closes #502 

I added a check if a seed is fixed for seqtk/sample when it samples paired-end reads.

I also added seqtk/subseq module and a test for it using existing test data.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: software/seqtk/sample/meta.yml [hunk 1]
```diff
@@ -36,4 +37,4 @@ output:
       pattern: "*.{fastq.gz}"
 
 authors:
-  - "@kaurravneet4123"
+  - "@kaurravneet4123,@sidorov-si"
```
Comment (drpatelh) [inline software/seqtk/sample/meta.yml:40]:
```suggestion
  - "@kaurravneet4123"
  - "@sidorov-si"
```

File: software/seqtk/subseq/main.nf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        container "quay.io/biocontainers/seqtk:1.3--h5bf99c6_3"
+    }
+
+    input:
+    path sequences
+    path filter_list
+
+    output:
+    path "*.gz",          emit: sequences
+    path "*.version.txt", emit: version
```
Comment (drpatelh) [inline software/seqtk/subseq/main.nf:27]:
```suggestion
    path "*.gz"         , emit: sequences
    path "*.version.txt", emit: version
```

File: software/seqtk/subseq/main.nf [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    script:
+    def software = getSoftwareName(task.process)
+    def ext = "fa"
+    if ("$sequences" ==~ /.+\.fq|.+\.fq.gz|.+\.fastq|.+\.fastq.gz/) {
```
Comment (drpatelh) [inline software/seqtk/subseq/main.nf:32]: Nice! This works?! 😆

File: software/seqtk/subseq/main.nf [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        ext = "fq"
+    }
+    """
+    seqtk \\
+        subseq \\
+        $options.args \\
+        $sequences \\
+        $filter_list | \\
+        gzip > ${sequences}_"filtered".${ext}.gz
```
Comment (drpatelh) [inline software/seqtk/subseq/main.nf:41]: Do we need to use `--no-name` like [here](https://github.com/nf-core/modules/pull/510/files#r638211785)?

File: software/seqtk/subseq/main.nf [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    input:
+    path sequences
+    path filter_list
+
+    output:
+    path "*.gz",          emit: sequences
+    path "*.version.txt", emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
```
Comment (drpatelh) [inline software/seqtk/subseq/main.nf:30]:
```suggestion
    def software = getSoftwareName(task.process)
    def prefix   = options.suffix ?: ''
```

File: software/seqtk/subseq/main.nf (hunk 4 above)
Comment (drpatelh) [inline software/seqtk/subseq/main.nf:41]:
```suggestion
        gzip > ${sequences}${prefix}.${ext}.gz
```

File: tests/software/seqtk/subseq/main.nf [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { SEQTK_SUBSEQ } from '../../../../software/seqtk/subseq/main.nf' addParams( options: [:] )
```
Comment (drpatelh) [inline tests/software/seqtk/subseq/main.nf:5]:
```suggestion
include { SEQTK_SUBSEQ } from '../../../../software/seqtk/subseq/main.nf' addParams( options: ['suffix':'.filtered'] )
```

File: tests/software/seqtk/subseq/test.yml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+## TODO nf-core: Please run the following command to build this file:
+#                nf-core modules create-test-yml seqtk/subseq
+- name: seqtk subseq
+  command: nextflow run ./tests/software/seqtk/subseq -entry test_seqtk_subseq -c tests/config/nextflow.config
+  tags:
+    - seqtk
+    - seqtk/subseq
+  files:
+    - path: output/seqtk/genome.fasta_filtered.fa.gz
```
Comment (drpatelh) [inline tests/software/seqtk/subseq/test.yml:9]:
```suggestion
    - path: output/seqtk/genome.fasta.filtered.fa.gz
```

File: tests/software/seqtk/subseq/test.yml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+## TODO nf-core: Please run the following command to build this file:
+#                nf-core modules create-test-yml seqtk/subseq
```
Comment (drpatelh) [inline tests/software/seqtk/subseq/test.yml:2]:
```suggestion
```

**Review by drpatelh — COMMENTED** [review]
Thanks @sidorov-si ! Just a few minor things to address.

File: software/seqtk/subseq/main.nf (hunk 3 above)
Comment (sidorov-si) [reply software/seqtk/subseq/main.nf:32]: Yup, I checked in the Groovy Playground :) -- https://groovy-playground.appspot.com/

File: software/seqtk/subseq/main.nf (hunk 4 above)
Comment (sidorov-si) [reply software/seqtk/subseq/main.nf]: Yes, exactly, thank you for the suggestion!

**Comment by sidorov-si** [comment]
> Looks like you have pulled in some of the commits recently pushed to `master` updating the `samtools` modules @sidorov-si ?

Yes, @drpatelh, probably I did: according to the README, I did `git pull --rebase upstream master` before continuing with my work on `seqtk`. Did I do something wrong?

**Comment by sidorov-si** [comment]
So now the module tests for `seqtk/sample` run successfully with Docker and Singularity (see the corresponding checks) but fail on Conda (I think, due to the randomiser in `seqtk/sample` producing a different series of pseudorandom numbers than in Docker/Singularity). `seqtk/subseq` passes all checks, and the only other failing check is for `samtools/view` on Conda (which I did not touch). Could I do anything else for this PR?

**Review by drpatelh — APPROVED** [review]
LGTM!

**Pull request merged**